File: 299-bulls-and-cows/bulls-and-cows.cpp
class Solution {
public:
    string getHint(string secret, string guess) {
        int x = 0, y = 0;
        map<char, int> m, mv;
        map<int,int> ind;
        int n = secret.size();
        for (int i = 0; i < n; i++) {
            m[secret[i]]++;
        }
        for (int i = 0; i < n; i++) {
            mv[guess[i]]++;
        }
        for (int i = 0; i < n; i++) {
            if (secret[i] == guess[i]) {
                x++;
                mv[guess[i]]--;
                m[guess[i]]--;
                ind[i]++;
            }
        }
        for (int i = 0; i < n; i++) {
            if (ind[i] == 0) {
                if (mv[guess[i]] > 0 && m[guess[i]] > 0) {
                    y++;
                    mv[guess[i]]--;
                    m[guess[i]]--;
                }
            }
        }
        return to_string(x) + 'A' + to_string(y) + 'B';
    }
};

File: 299-bulls-and-cows/bulls-and-cows.java
class Solution {
    public String getHint(String secret, String guess) {
        int n=secret.length();
        int a=0,b=0;
        int[] mp = new int[10];
        for(int i=0;i<n;i++)
        {
            Character ch = secret.charAt(i);
            mp[ch-'0']++;
        }
        for(int i=0;i<n;i++)
        {
            Character ch = secret.charAt(i);
            Character ch2 = guess.charAt(i);
            if(ch==ch2)
            {
                a++;
                 mp[ch-'0']--;
            }
        }
        for(int i=0;i<n;i++)
        {
            Character ch = secret.charAt(i);
            Character ch2 = guess.charAt(i);
           
                if(ch!=ch2 && mp[ch2-'0']!=0)
                {
                    b++;
                     mp[ch2-'0']--;
                }
        }
        String p = a+"A"+b+"B";
        return p;
    }
}